entity-service/src/main/kotlin/com/egm/stellio/entity/repository/Neo4jRepository.kt

```diff
@@ -24,30 +23,38 @@ class Neo4jRepository(
     private val session: Session,
     private val sessionFactory: SessionFactory
 ) {
-    private val logger = LoggerFactory.getLogger(javaClass)
 
-    /**
-     * Create the concrete relationship from a relationship to an entity.
-     *
-     * It is used for :
-     *   - relationships of relationships
-     *   - relationships of properties
-     */
-    @Transactional
-    fun createRelationshipToEntity(relationshipId: String, relationshipType: String, entityId: String): Int {
+    fun createPropertyOfSubject(subjectId: String, property: Property): String {
         val query = """
-            MATCH (subject:Attribute { id: "$relationshipId" }), (target:Entity { id: "$entityId" })
-            MERGE (subject)-[:$relationshipType]->(target)
-            MERGE (subject)-[:HAS_OBJECT]->(target)
-            """
+            MATCH (subject { id: '$subjectId' })
+            CREATE (subject)-[:HAS_VALUE]->(p:Attribute:Property ${'$'}props)
+            RETURN p.id as id
+        """
+
+        return session.query(query, mapOf("props" to property.nodeProperties()))
+            .toList()
+            .map { it["id"] as String }
+            .first()
+    }
 
-        return session.query(query, emptyMap<String, String>()).queryStatistics().relationshipsCreated
+    fun createRelationshipOfSubject(subjectId: String, relationship: Relationship, targetId: String): String {
+        val relationshipType = relationship.type[0].toRelationshipTypeName()
+        val query = """
+            MATCH (subject { id: '$subjectId' }), (target:Entity { id: '$targetId' })
+            CREATE (subject)-[:HAS_OBJECT]->(r:Attribute:Relationship:`${relationship.type[0]}` ${'$'}props)-[:HAS_OBJECT]->(target)
+            CREATE (r)-[:$relationshipType]->(target)
+            RETURN r.id as id
+        """
+
+        return session.query(query, mapOf("props" to relationship.nodeProperties()))
+            .toList()
+            .map { it["id"] as String }

```

Comment:
Better to get first() then to get ["id"] as String or i'm wrong?

Reply:
Yes indeed